File: src/main/java/org/cloudfoundry/credhub/entity/ValueCredentialVersionData.java
package org.cloudfoundry.credhub.entity;

import javax.persistence.DiscriminatorValue;
import javax.persistence.Entity;

@Entity
@DiscriminatorValue(ValueCredentialVersionData.CREDENTIAL_TYPE)
public class ValueCredentialVersionData extends CredentialVersionData<ValueCredentialVersionData> {

  public static final String CREDENTIAL_TYPE = "value";

  public ValueCredentialVersionData() {
  }

  public ValueCredentialVersionData(String name) {
    super(name);
  }

  @Override
  public String getCredentialType() {
    return CREDENTIAL_TYPE;
  }
}

File: components/credentials/src/main/kotlin/org/cloudfoundry/credhub/entity/ValueCredentialVersionData.kt
package org.cloudfoundry.credhub.entity

import jakarta.persistence.DiscriminatorValue
import jakarta.persistence.Entity

@Entity
@DiscriminatorValue(ValueCredentialVersionData.CREDENTIAL_TYPE)
class ValueCredentialVersionData : CredentialVersionData<ValueCredentialVersionData> {
    override val credentialType: String
        get() = CREDENTIAL_TYPE

    constructor() : super()

    constructor(name: String) : super(name)

    companion object {
        const val CREDENTIAL_TYPE = "value"
    }
}
